python/ray/serve/controller.py

```diff
@@ -320,37 +323,84 @@ async def run_control_loop(self) -> None:
 
             # Update the active nodes set before updating the HTTP states, so they
             # are more consistent.
+            node_update_start_time = time.time()
             self._update_active_nodes()
+            self.node_update_duration_gauge.set(time.time() - node_update_start_time)
 
             # Don't update http_state until after the done recovering event is set,
             # otherwise we may start a new HTTP proxy but not broadcast it any
             # info about available deployments & their replicas.
             if self.http_proxy_state_manager and self.done_recovering_event.is_set():
                 try:
+                    proxy_update_start_time = time.time()
                     self.http_proxy_state_manager.update(
                         active_nodes=self._active_nodes
                     )
+                    self.proxy_update_duration_gauge.set(
+                        time.time() - proxy_update_start_time
+                    )
                 except Exception:
                     logger.exception("Exception updating HTTP state.")
 
             try:
+                dsm_update_start_time = time.time()
                 any_recovering = self.deployment_state_manager.update()
+                self.dsm_update_duration_gauge.set(time.time() - dsm_update_start_time)
                 if not self.done_recovering_event.is_set() and not any_recovering:
                     self.done_recovering_event.set()
             except Exception:
                 logger.exception("Exception updating deployment state.")
 
             try:
+                asm_update_start_time = time.time()
                 self.application_state_manager.update()
+                self.asm_update_duration_gauge.set(time.time() - asm_update_start_time)
             except Exception:
                 logger.exception("Exception updating application state.")
 
             try:
+                snapshot_start_time = time.time()
                 self._put_serve_snapshot()
+                self.snapshot_duration_gauge.set(time.time() - snapshot_start_time)
             except Exception:
                 logger.exception("Exception putting serve snapshot.")
+            loop_duration = time.time() - loop_start_time
+            if loop_duration > 60:
```

Comment:
60 is an extremely high threshold. suggest to do 10s

but either way let's check that the chosen threshold makes sense in your load tests

Reply:
Sounds good, I switched to 10s.